### PR TITLE
Check balance first before wait for Transfer

### PR DESCRIPTION
**Pull request by pzhao5**
If balance is there, then don't need to wait further for the transfer

### CODE REVIEW

File: services/eth_gateway.go [hunk 1]
```diff
@@ -266,6 +266,12 @@ func getCurrentBlock() (*types.Block, error) {
 // WaitForTransfer is blocking call that will observe on brokerAddr on transfer on ETH.
 // If it is completed return number of PRL.
 func waitForTransfer(brokerAddr common.Address) (*big.Int, error) {
+	balance := checkBalance(brokerAddr)
+	if balance.Int64() > 0 {
+		// Has balance already, don't need to wait for it.
+		return balance, nil
+	}
```
Comment (rfornea) [inline services/eth_gateway.go]: Does checkBalance return the balance of PRL or the balance of ETH?  Or both?

Comment (pzhao5) [reply services/eth_gateway.go]: Ideally, it should be PRL, and we use it in the codebase. Maybe Astor could have a better answer

Comment (AaronVasquez) [reply services/eth_gateway.go]: @Astor ^

Comment (rfornea) [reply services/eth_gateway.go]: Ultimately we will need methods to get balances of both ETH and PRL. I created a Zenhub issue to have methods for both.   

If everyone is cool with it we could just go ahead and approve this, and splitting out those methods could be something we do soonish. We will have to look through the codebase at where the current method is being used and make sure it is replaced with the correct one.

Comment (Astor) [reply services/eth_gateway.go]: It checks the for both balances.

**Pull request merged**